packages/transition-frontend/src/components/forms/comparison/ScenarioComparisonPanel.tsx

```diff
@@ -295,7 +295,11 @@ class ScenarioComparisonPanel extends ChangeEventsForm<ComparisonPanelProps, Com
     private getScenarioNameById(id: string): string {
         for (const scenario of this.state.scenarioCollection.features) {
             if (scenario.id === id) {
-                return scenario.toString(false);
+                let scenarioName = scenario.toString(false) as string;
+                if (scenarioName.length >= 30) {
```

Comment:
Can you add a comment that says something like: 
// We are not simply doing a substr(0,30) as we want to add "..." when we have to cut it down